### PR TITLE
Fix check for no affinitization in GC on Unix

### PR DESCRIPTION
**Pull request by janvorli**
While the affinitization mask setting is not used for affinity setting
on Unix, it was still being checked for non-zero as one of the
indicators that GC heap affinity was specified. This change removes
that check for Unix builds.

### CODE REVIEW

File: src/gc/gc.cpp [hunk 1]
```diff
@@ -34781,7 +34781,11 @@ HRESULT GCHeap::Initialize()
         return CLR_E_GC_BAD_AFFINITY_CONFIG;
     }
 
-    if ((cpu_index_ranges_holder.Get() != nullptr) || (config_affinity_mask != 0))
+    if ((cpu_index_ranges_holder.Get() != nullptr)
+#ifndef FEATURE_PAL
```
Comment (jkotas) [inline src/gc/gc.cpp]: Should this rather be `PLATFORM_WINDOWS` ? Note that `FEATURE_PAL` is not defined for standalone GC.

Comment (janvorli) [reply src/gc/gc.cpp]: You are right - and we have yet another FEATURE_PAL ifdef in the gc that should get the same treatment.

**Comment by janvorli** [comment]
I've changed the ifdef from FEATURE_PAL to PLATFORM_WINDOWS (both in my change and the pre-existing one)

**Comment by Maoni0** [comment]
>         gc_heap::gc_thread_no_affinitize_p = ((config_affinity_set.Count() == 0) && (config_affinity_mask == 0));

this should be replaced by affinity_config_specified_p. 

---
Refers to: src/gc/gc.cpp:34812 in 40b8a91. [](commit_id = 40b8a9169069e1dc6f4145132210d27b5e877cea, deletion_comment = False)

**Pull request merged**